massa-bootstrap/src/server.rs

```diff
@@ -224,23 +227,18 @@ impl BootstrapServer<'_> {
         interval: Duration,
         stopper: Receiver<()>,
     ) -> Result<(), Box<BootstrapError>> {
+        let ticker = tick(interval);
+
         loop {
-            std::thread::sleep(interval);
-            match stopper.try_recv() {
-                Ok(()) => return Ok(()),
-                Err(e) => match e {
-                    crossbeam::channel::TryRecvError::Empty => {}
-                    crossbeam::channel::TryRecvError::Disconnected => {
-                        return Err(BootstrapError::GeneralError(
-                            "update stopper unexpected disconnect".to_string(),
-                        )
-                        .into());
+            select! {
+                recv(stopper) -> res => {
+                    match res {
+                        Ok(()) => return Ok(()),
+                        Err(e) => return Err(Box::new(BootstrapError::GeneralError(format!("update stopper error : {}", e)))),
                     }
                 },
+                recv(ticker) -> _ => {list.update()?;},
```

Comment:
add a try_recv on the stopper here: if both are ready, then it's not guaranteed that it will choose the stopper.

In this scenario, the reality is that it's almost never gonna happen. The value will be more in self-documenting the code: "if we _can_ recieve on the stopper, that takes ultimate priority"

Reply:
in fact, priority is not specialy needed here. it's better but not needed. 
it's works well for now and there is no more waiting time for stop

Reply:
reasonably put. resolved.